pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -224,14 +233,27 @@ func readEntity(rows *sql.Rows, r FieldSelectRequest) (*entity.Entity, error) {
 }
 
 func (s *sqlEntityServer) Read(ctx context.Context, r *entity.ReadEntityRequest) (*entity.Entity, error) {
+	ctx, span := s.tracer.Start(ctx, "storage_server.Read")
+	defer span.End()
+	span.SetAttributes(attribute.String("request", r.String()))
+	s.log = s.log.FromContext(ctx)
```

Comment:
i don't think i've used `log.FromContext`, but my fuzzy understanding was that it might include values that are only relevant to that context (e.g. trace ID). is that correct? 

if so, it doesn't seem like `s.log` should be changed since those values might be ephemeral:
```suggestion
	log := s.log.FromContext(ctx)
```

Reply:
Yeah, the only reason I'm using it is to pass the traceID to the logs. Good point - I'll update that 👍

Reply:
I'm still on the fence about how to do this properly, or if the current way will cause a problem. I shouldn't have to create a new logger in every function 🤔

Reply:
it seems to be what [k8s does](https://github.com/search?q=repo%3Akubernetes%2Fkubernetes+klog.FromContext&type=code)

Reply:
Yeah, it was causing concurrency problems. This change fixed it 👍